vmm/src/vm.rs

```diff
@@ -1616,6 +1720,41 @@ impl<'a> Vm<'a> {
         // Unblock all CPU threads.
         vcpu_thread_barrier.wait();
 
+        let mut signals: [i32; MAX_SIGNAL_VAL] = [0; MAX_SIGNAL_VAL];
+        let mut index = 0;
+        for sig_num in 1..MAX_SIGNAL_VAL {
+            if self.devices.signal_present[sig_num] {
+                signals[index] = sig_num as i32;
+                index += 1;
+            }
+        }
+        if index > 0 {
+            if let Some(console_input) = &self.devices.console_input {
+                let console_input_clone = console_input.clone();
+                set_handler(&signals[0..index], move |received_signals| {
+                    for signal in received_signals.iter() {
+                        if *signal == SIGWINCH {
```

Comment:
Could the code for querying and updating be put into console.rs. I see there is similar code in ::new()

Reply:
As per suggestion from Sebastien, we should only do it from vm.rs. So, instead of moving this code to console.rs, I moved everything to vm.rs.